src/main/java/org/aksw/simba/lemming/creation/Inferer.java

```diff
@@ -239,27 +258,27 @@ public void iterateStmts(Model newModel, Model sourceModel, OntModel ontModel, M
 		OntProperty property = null;
 
 		if (node != null) {
-			property = (OntProperty) node.getAttribute();
+			property = node.getAttribute();
 			Property newPredicate = ResourceFactory.createProperty(node.getName());
 
 			if (!newPredicate.getURI().equals(predicate.getURI()))
-				ModelUtil.replaceStatement(newModel, statement,
+				ModelUtil.replaceStatement(model, statement,
 						ResourceFactory.createStatement(subject, newPredicate, object));
 		}
 
 		if (property != null) {
 			List<? extends OntResource> domain = property.listDomain().toList();
 			for (OntResource curResource : domain) {
 				Statement subjType = ResourceFactory.createStatement(subject, RDF.type, curResource);
-				if (!newModel.contains(subjType) && !curResource.isAnon()) {
+				if (!model.contains(subjType) && !curResource.isAnon()) {
 					newStmts.add(subjType);
 				}
 			}
 			if (object.isResource()) {
 				List<? extends OntResource> range = property.listRange().toList();
 				for (OntResource curResource : range) {
 					Statement objType = ResourceFactory.createStatement(object.asResource(), RDF.type, curResource);
-					if (!newModel.contains(objType)) {
+					if (!model.contains(objType)) {

```

Comment:
Same thing as the comment regarding `model.contains(stmt);`

Reply:
Done